### PR TITLE
Hide empty columns on new mod select design

### LINKED ISSUE #18154: Mod select overlay should not show empty columns
### Type

Cosmetic

### Bug description

This was touched on previously in the [integration PR's contention points](https://github.com/ppy/osu/pull/18111), and it's one which could get bad quickly when, for example, joining a match which has freemods limited to the last/fun column in the overlay, as pointed out in the video below.

### Screenshots or videos

https://user-images.githubusercontent.com/22781491/167294360-c20b76a4-c95c-4648-b320-10649110d111.mp4

### Version

master

### Logs

-

### PR DESCRIPTION
**Pull request by bdach**
Closes #18154.

https://user-images.githubusercontent.com/20418176/167300827-ac083d43-c168-483e-a458-828939686c65.mp4

This immediately exposes more issues, most of which are mentioned [in this discord conversation](https://discord.com/channels/188630481301012481/188630652340404224/972854220292251678), but I'm PRing at least for discussion/so that it can be tweaked with further.

As for the code itself, the most annoying part was preserving the pop in/out animations wherein columns fly in from alternate directions. I have managed to salvage it, but I'm not sure if the end result is legible.

### CODE REVIEW

**Comment by frenzibyte** [comment]
Haven't deeply checked that, but I'm wondering if the pop in/out changes could be simplified using `columnsFlow.Where(c => !c.Column.AllFiltered.Value)` instead?

**Comment by bdach** [comment]
That logic was written in a very particular way to avoid issues with scenarios such as:

* Set filter once
* Hide overlay
* Set filter again
* Show overlay

The issue you want to avoid is that for instance, if a column didn't pop out at the hide (so it didn't move off the screen), but then pops in and already is at the correct Y position, it wouldn't end up moving in at all. Always applying transitions, but doing so instantaneously on filtered and with duration on non-filtered columns avoids this.

File: osu.Game/Overlays/Mods/ModSelectScreen.cs [hunk 1]
```diff
@@ -362,13 +372,26 @@ protected override void PopIn()
...
             {
-                columnFlow[i].Column
-                             .TopLevelContent
-                             .Delay(i * 30)
-                             .MoveToY(0, fade_in_duration, Easing.OutQuint)
-                             .FadeIn(fade_in_duration, Easing.OutQuint);
+                var column = columnFlow[i].Column;
+
+                double delay = column.AllFiltered.Value ? 0 : nonFilteredColumnCount * 30;
+                double duration = column.AllFiltered.Value ? 0 : fade_in_duration;
```
Comment (smoogipoo) [inline osu.Game/Overlays/Mods/ModSelectScreen.cs:382]: Is there a reason why this whole code is running if `column.AllFiltered == true`? That is, is there a reason why this couldn't be a precondition:
```
for (...)
{
    if (column.AllFiltered.Value)
        continue;

    ...

    nonFilteredColumnCount += 1;
}
```

Comment (bdach) [reply osu.Game/Overlays/Mods/ModSelectScreen.cs:382]: I've already touched on why this is being done this way [above](https://github.com/ppy/osu/pull/18157#issuecomment-1120437194).

**Review by smoogipoo — APPROVED** [review]

**Comment by peppy** [comment]
As I've mentioned in discord, from a UX perspective I'm not sure I agree with this. But let's push this out and see what users say, if anything.

**Review by peppy — APPROVED** [review]

**Pull request merged**